schematics/types/compound.py

```diff
@@ -196,26 +194,23 @@ def export_loop(self, list_instance, field_converter,
             if hasattr(self.field, 'export_loop'):
                 shaped = self.field.export_loop(value, field_converter,
                                                 role=role)
-                feels_empty = shaped and len(shaped) == 0
+                feels_empty = shaped is None or len(shaped) == 0
             else:
```

Comment:
Basically I was looking for the original thought behind this line.

`feels_empty` has always been `False` here because in no situation would `bool(shaped) == True` and `len(shaped) == 0` occur simultaneously.

Still looks good to me, and this is necessary preparation for #351 and the undefined feature, so merging.